caikit/runtime/http_server.py

```diff
@@ -179,11 +183,11 @@ def shutdown_wrapper(*args, **kwargs):
         self._uvicorn_server_thread = None
 
     def __del__(self):
-        if get_config().runtime.metering.enabled:
+        if get_config() and get_config().runtime.metering.enabled:
```

Comment:
Why do we need the truthiness check on `if get_config()`? Are there cases you've encountered where this results in an empty dict or `None`? If we _do_ need that check, we should only call `get_config()` once since it _might_ do some work in there some day.

```suggestion
        config = get_config()
        if config and config.runtime.metering.enabled:
```